### DISCUSSION TITLE
failed on django-admin runserver --settings=example.settings

### DISCUSSION THREAD

**Discussion by SeekPoint** [django-json-api/django-rest-framework-json-api#1106] [Q&A] [answered]
(venv) PS C:\django-rest-framework-json-api> django-admin runserver --settings=example.settings
C:\django-rest-framework-json-api\venv\Scripts\python.exe: Error while finding module specification for '__main__' (ValueError: __main__.__spec__ is None)
(venv) PS C:\django-rest-framework-json-api>

**Comment by sliverc** [comment, marked as answer]
Thanks for reporting this issue. I unfortunately cannot reproduce this issue. It might be an issue with a corrupted venv. As #1103 is now fixed could you try again? Best delete your virtual env first and follow the steps in the README again to see whether this issue comes up again.

**Comment by SeekPoint** [comment]
the issue exists yet!

windows 11

C:\django-rest-framework-json-api\venv\Scripts\python.exe: Error while finding module specification for '__main__' (ValueError: __main__.__spec__ is None)
(venv) PS C:\django-rest-framework-json-api>


(venv) PS C:\django-rest-framework-json-api> python
Python 3.9.0 (tags/v3.9.0:9cf6752, Oct  5 2020, 15:34:40) [MSC v.1927 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license" for more information.
>>> quit()
(venv) PS C:\django-rest-framework-json-api> pip list
Package                       Version   Editable project location
----------------------------- --------- ---------------------------------
alabaster                     0.7.12
asgiref                       3.5.2
attrs                         22.1.0
Babel                         2.11.0
black                         22.10.0
bleach                        5.0.1
certifi                       2022.9.24
charset-normalizer            2.1.1
click                         8.1.3
colorama                      0.4.6
colored                       1.4.4
commonmark                    0.9.1
coverage                      6.5.0
Django                        4.1.3
django-filter                 22.1
django-polymorphic            3.1.0
djangorestframework           3.14.0
djangorestframework-jsonapi   6.0.0     c:\django-rest-framework-json-api
docutils                      0.17.1
exceptiongroup                1.0.3
factory-boy                   3.2.1
Faker                         15.2.0
flake8                        5.0.4
flake8-bugbear                22.10.27
flake8-isort                  5.0.0
idna                          3.4
imagesize                     1.4.1
importlib-metadata            5.0.0
inflection                    0.5.1
iniconfig                     1.1.1
isort                         5.10.1
jaraco.classes                3.2.3
Jinja2                        3.1.2
keyring                       23.11.0
MarkupSafe                    2.1.1
mccabe                        0.7.0
more-itertools                9.0.0
mypy-extensions               0.4.3
packaging                     21.3
pathspec                      0.10.2
pip                           22.3.1
pkginfo                       1.8.3
platformdirs                  2.5.4
pluggy                        1.0.0
pycodestyle                   2.9.1
pyflakes                      2.5.0
Pygments                      2.13.0
pyparsing                     3.0.9
pytest                        7.2.0
pytest-cov                    4.0.0
pytest-django                 4.5.2
pytest-factoryboy             2.5.0
python-dateutil               2.8.2
pytz                          2022.6
pywin32-ctypes                0.2.0
PyYAML                        6.0
readme-renderer               37.3
recommonmark                  0.7.1
requests                      2.28.1
requests-toolbelt             0.10.1
rfc3986                       2.0.0
rich                          12.6.0
setuptools                    65.5.1
six                           1.16.0
snowballstemmer               2.2.0
Sphinx                        5.3.0
sphinx-rtd-theme              1.1.1
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
sqlparse                      0.4.3
syrupy                        3.0.4
tomli                         2.0.1
twine                         4.0.1
typing_extensions             4.4.0
tzdata                        2022.6
uritemplate                   4.1.1
urllib3                       1.26.12
webencodings                  0.5.1
wheel                         0.37.1
zipp                          3.10.0
(venv) PS C:\django-rest-framework-json-api>

**Comment by sliverc** [comment]
When looking at the error, I have a feeling that this has something to do with your setup and not with the example app itself. Reason being is that in the example app there is no __main__ defined which the error shows. The main is part of django.

I therefore converted this to a discussion. I do not use Windows, so do not know what could cause this error. What I would try (if you haven't tried it yet) is one of the following:

1. Delete the virtualenv and create a new one again
2. Try to use [anaconda](https://docs.anaconda.com/navigator/index.html) for handling Python and virtualenv which might solve your problem.

Maybe other users who use Windows with DJA might be able to assist you as well. If someone else can reproduce this issue, please comment here as well.

**Comment by SeekPoint** [comment]
I am using virtualenv

**Comment by miazocool** [comment]
Hello, i am experiencing the same issue using Windows 10, i have deleted and created a new the venv, but still throws the error
